cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go

```diff
@@ -115,6 +115,13 @@ func (aws *awsCloudProvider) NodeGroupForNode(node *apiv1.Node) (cloudprovider.N
 		klog.Warningf("Node %v has no providerId", node.Name)
 		return nil, nil
 	}
+
+	// Skip SageMaker instances
+	if strings.HasPrefix(node.GetName(), "hyperpod") {
```

Comment:
Instead of prefix on the node name, can we rely on a custom tag on the node?

Reply:
Sure, the aws hyperpod engineer @surajkota suggests me to check the prefix in order to match the current pattern for fargate so I modified it to the current approach that comparing the prefix. 

https://github.com/kubernetes/autoscaler/blob/master/cluster-autoscaler/cloudprovider/aws/aws_cloud_provider.go#L142-L144
https://github.com/kubernetes/autoscaler/pull/8195#discussion_r2127350692